randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -1,12 +1,227 @@
+        return {
+            "save_room": self.game.region_list.node_by_identifier(self.patches.starting_location).extra["save_room"]
+        }
+
+    def _create_hash_dict(self):
+        return {
+            "word_hash": self.description.shareable_word_hash,
+            "hash": self.description.shareable_hash
+        }
+
+    def _create_game_patches(self):
+        game_patches = {
+            "septogg_helpers": self.patches.configuration.septogg_helpers,
+            "respawn_bomb_blocks": self.patches.configuration.respawn_bomb_blocks,
+            "skip_cutscenes": self.patches.configuration.skip_cutscenes,
+            "energy_per_tank": self.patches.configuration.energy_per_tank,
+            "grave_grotto_blocks": self.patches.configuration.grave_grotto_blocks,
+            "fusion_mode": self.patches.configuration.fusion_mode,
+            "nest_pipes": self.patches.configuration.nest_pipes,
+            "softlock_prevention_blocks": self.patches.configuration.softlock_prevention_blocks,
+            "a3_entrance_blocks": self.patches.configuration.a3_entrance_blocks,
+            "screw_blocks": self.patches.configuration.screw_blocks,
+        }
+        for item, state in self.patches.configuration.ammo_pickup_configuration.pickups_state.items():
+            launcher_text = ""
+            if item.name == "Missile Expansion":
+                launcher_text = "require_missile_launcher"
+            elif item.name == "Super Missile Expansion":
+                launcher_text = "require_super_launcher"
+            elif item.name == "Power Bomb Expansion":
+                launcher_text = "require_pb_launcher"
+
+            if not launcher_text:
+                continue
+
+            game_patches[launcher_text] = state.requires_main_item
+        return game_patches
+
+    def _create_door_locks(self):
+        locks = {}
+        for node, weakness in self.patches.all_dock_weaknesses():
+            locks[str(node.extra["instance_id"])] = {
+                "lock": weakness.long_name
+            }
+        return locks
+
+    def _create_hints(self):
+        artifacts = [self.game.resource_database.get_item(f"Metroid DNA {i + 1}") for i in range(46)]
+        ice = [(self.game.resource_database.get_item("Ice Beam"))]
+        artifact_hints = {}
+        hint_config = self.patches.configuration.hints
+        if hint_config.artifacts != ItemHintMode.DISABLED:
+            artifact_hints = guaranteed_item_hint.create_guaranteed_hints_for_resources(
+                self.description.all_patches,
+                self.players_config,
+                AM2RHintNamer(self.description.all_patches, self.players_config),
+                hint_config.artifacts == ItemHintMode.HIDE_AREA,
+                artifacts,
+                False  # TODO: set this to true, when patcher supports setting colors!
+            )
+        else:
+            artifact_hints = {k: f"{k.long_name} is hidden somewhere on SR-388." for k in artifacts}
```

Comment:
you know, this isn't just an am2r problem but it occurs to mii that these disabled hint texts are just straight up lying in cross-game multiworld lol. nbd here since it's a problem in other games too, but somewhat funny

Reply:
AFAIK this is technically true for dread and am2r, since DNA is always on the same world. May change for am2r later, will try to think of another funny description if that happens.

Reply:
Whoops, fixed locally